codec/api/json.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Gin Core Team. All rights reserved.
+// Use of this source code is governed by a MIT style
+// license that can be found in the LICENSE file.
+
+package api
+
+import "io"
+
+// JsonApi the api for json codec.
+type JsonApi interface {
+	Marshal(v any) ([]byte, error)
+	Unmarshal(data []byte, v any) error
+	MarshalIndent(v any, prefix, indent string) ([]byte, error)
+	NewEncoder(writer io.Writer) JsonEncoder
+	NewDecoder(reader io.Reader) JsonDecoder
+}
+
```

Comment:
1. Should we move the file to `json` folder?
2. Rename the `JsonApi` to `API`.
3. `JsonEncoder` => `Encoder`
4. `JsonDecoder` => `Decoder`

Reply:
Hi, @appleboy . I have refacted the PR. PTAL, Thanks.

Reply:
@timandy LGTM